### ISSUE TITLE
Test: obj_critnib_mt/TEST2: SETUP (all/pmem/static-debug/drd) fails

### ISSUE THREAD

**Issue by Greg091** [pmem/issues#1040]
# ISSUE: 

## Environment Information 

- PMDK package version(s): 1.6-rc2-1-gfe22f9065
- OS(es) version(s): Fedora29
- ndctl version(s): 64.1.1.g43bf086
- kernel version(s): 4.20.16

## Please provide a reproduction of the bug:

./RUNTESTS obj_critnib_mt -s TEST2 -t all -f pmem

## Actual behavior:
```
$ ./RUNTESTS obj_critnib_mt -s TEST2 -t all -f pmem
obj_critnib_mt/TEST2: SETUP (all/pmem/debug/drd)
obj_critnib_mt/TEST2 failed with Valgrind. See drd2.log. Last 20 lines below.
obj_critnib_mt/TEST2 drd2.log ==44481==    by 0x405BD8: util_mutex_unlock (sys_util.h:119)
obj_critnib_mt/TEST2 drd2.log ==44481==    by 0x406530: critnib_insert (critnib.c:398)
obj_critnib_mt/TEST2 drd2.log ==44481==    by 0x4052F4: thread_write1024 (obj_critnib_mt.c:131)
obj_critnib_mt/TEST2 drd2.log ==44481==    by 0x483EEA4: vgDrd_thread_wrapper (drd_pthread_intercepts.c:444)
obj_critnib_mt/TEST2 drd2.log ==44481==    by 0x497458D: start_thread (in /usr/lib64/libpthread-2.28.so)
obj_critnib_mt/TEST2 drd2.log ==44481==    by 0x4A8B6A2: clone (in /usr/lib64/libc-2.28.so)
obj_critnib_mt/TEST2 drd2.log ==44481== Other segment end (thread 36)
obj_critnib_mt/TEST2 drd2.log ==44481==    at 0x48421B8: pthread_mutex_lock_intercept (drd_pthread_intercepts.c:888)
obj_critnib_mt/TEST2 drd2.log ==44481==    by 0x48421B8: pthread_mutex_lock (drd_pthread_intercepts.c:898)
obj_critnib_mt/TEST2 drd2.log ==44481==    by 0x40F920: os_mutex_lock (os_thread_posix.c:123)
obj_critnib_mt/TEST2 drd2.log ==44481==    by 0x405B83: util_mutex_lock (sys_util.h:87)
obj_critnib_mt/TEST2 drd2.log ==44481==    by 0x4063A1: critnib_insert (critnib.c:362)
obj_critnib_mt/TEST2 drd2.log ==44481==    by 0x4052F4: thread_write1024 (obj_critnib_mt.c:131)
obj_critnib_mt/TEST2 drd2.log ==44481==    by 0x483EEA4: vgDrd_thread_wrapper (drd_pthread_intercepts.c:444)
obj_critnib_mt/TEST2 drd2.log ==44481==    by 0x497458D: start_thread (in /usr/lib64/libpthread-2.28.so)
obj_critnib_mt/TEST2 drd2.log ==44481==    by 0x4A8B6A2: clone (in /usr/lib64/libc-2.28.so)
obj_critnib_mt/TEST2 drd2.log ==44481==
obj_critnib_mt/TEST2 drd2.log ==44481==
obj_critnib_mt/TEST2 drd2.log ==44481== For counts of detected and suppressed errors, rerun with: -v
obj_critnib_mt/TEST2 drd2.log ==44481== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 12 from 8)
RUNTESTS: stopping: obj_critnib_mt/TEST2 failed, TEST=all FS=pmem BUILD=debug
1 tests failed:
obj_critnib_mt/TEST2
```
Logs:
[pmem2.log](https://github.com/pmem/issues/files/2977677/pmem2.log)
[trace2.log](https://github.com/pmem/issues/files/2977678/trace2.log)
[drd2.log](https://github.com/pmem/issues/files/2977679/drd2.log)
[out2.log](https://github.com/pmem/issues/files/2977680/out2.log)

## Expected behavior:

The test should pass.

## Additional information about Priority and Help Requested:

Are you willing to submit a pull request with a proposed change? (Yes, No)  

Requested priority: (Showstopper, High, Medium, Low)

**Comment by kilobyte** [comment]
"often" is not quite the case: I've so far managed to reproduce this bug once, despite trying on many machines; the one that "succeeded" is Debian unstable (valgrind 3.14), after ~100 tries.

This'll make debugging and verifying a fix unfun.

**Comment by kilobyte** [comment]
Merged to 1.6.

**Comment by Greg091** [comment]
Verified on: 1.6-rc2-15-g0a0d0821a
